src/parquet.cc

```diff
@@ -33,8 +33,20 @@ using parquet::PageType;
 using parquet::SchemaElement;
 using parquet::Type;
 
-InMemoryInputStream::InMemoryInputStream(const uint8_t* buffer, int64_t len) :
-  buffer_(buffer), len_(len), offset_(0) {
+InMemoryInputStream::InMemoryInputStream(int64_t len) : len_(len), offset_(0) {
+  buffer_ = (uint8_t*)malloc(sizeof(uint8_t) * len_);
+}
+
+InMemoryInputStream::~InMemoryInputStream() {
+  free(buffer_);
+}
```

Comment:
I think it would be preferable to leave the management of this memory to parquet_reader.cc; the intent of this class was to implement InputStream on any contiguous data in virtual address space, so by removing the `buffer` parameter it is no longer able to work with memory regions that are not allocated and freed by the class instance.